File: todolist/src/main/java/com/projects/todolist/service/TaskService.java
package com.projects.todolist.service;

import com.projects.todolist.dto.TaskDto;
import java.util.List;

public interface TaskService {
	
	TaskDto addTask(TaskDto taskDto);

	TaskDto getTask(Long id);

    List<TaskDto> getAllTasks();

    TaskDto updateTask(TaskDto taskDto, Long id);

    void deleteTask(Long id);

    TaskDto completeTask(Long id);

    TaskDto incompleteTask(Long id);

}

File: react-todolist/src/services/TaskService.js
import axios from 'axios'

const BASE_REST_API_URL = 'http://localhost:8080/api/tasks'

export const getAllTasks = () => axios.get(BASE_REST_API_URL)

export const saveTask = (task) => axios.post(BASE_REST_API_URL, task)

export const getTask = (id) => axios.get(BASE_REST_API_URL + '/' + id)

export const updateTask = (id, task) => axios.put(BASE_REST_API_URL + '/' + id, task)

export const deleteTask = (id) => axios.delete(BASE_REST_API_URL + '/' + id)

export const completeTask = (id) => axios.patch(BASE_REST_API_URL + '/' + id + '/complete')

export const incompleteTask = (id) => axios.patch(BASE_REST_API_URL + '/' + id + '/incomplete')
